### PR TITLE
Add support for Multilingualization (m17n)

### PR DESCRIPTION
**Pull request by UuuNyaa**
I create a pull request for Multilingualization (m17n) support.
I want native (Japanese and Simplified Chinese) speakers to review it.

Of course, you are welcome to provide translation resources in other languages.

### CODE REVIEW

**Comment by UuuNyaa** [comment]
The following URL is a good reference for translation.
https://github.com/blender/blender-translations/tree/master/po

**Comment by nagadomi** [comment]
If powroupi will merge this PR, I can revise the Japanese translation.

My personal concern is that translations without `text_ctxt` option will affect blender itself and other addons as well.
However, it is not practical to add `text_ctxt` option to all `label()`, `operator()` etc :(

Regarding the Japanese translation, there are no errors at the natural language layer, but I think it would be better to use the same name for the same field as in PMX Editor. (If you only have a Linux machine, you can run PMX Editor on wine 32bit.)
For example, 表示パネル→表示枠, 両面表示→両面描画, リジッドボティ→剛体, etc.
Some names already have a translation in blender itself and may not be overridable. This also relates to `text_ctxt` option.

**Comment by UuuNyaa** [comment]
@nagadomi Thanks for your comment.

> I think it would be better to use the same name for the same field as in PMX Editor.

It is a difficult decision.
I guess it would be less confusing for users if we adopt Blender's vocabularies.
Furthermore, PMX Editor vocabularies cannot be referenced in translations of other languages, such as Chinese.

**Comment by powroupi** [comment]
@UuuNyaa, thank you, the changes looks great, and I think I'll merge this PR when your changes are all done. Thank you very much. :smile:

>My personal concern is that translations without text_ctxt option will affect blender itself and other addons as well.

I'm not sure how translations affect other UI. Shouldn't it only affect the UI of `mmd_tools` since we register it with our addon's name? :thinking:

File: mmd_tools/__init__.py [hunk 1]
```diff
@@ -144,7 +144,18 @@ def register():
...
+        from mmd_tools.m17n import translation_dict
+        bpy.app.translations.register(bl_info['name'], translation_dict)
+    except:
+        pass
+
 def unregister():
+    try:
+        bpy.app.translations.unregister(bl_info['name'])
+    except:
+        pass
```
Comment (powroupi) [inline mmd_tools/__init__.py]: I think it is better to remove `try...except...` unless we know which errors could be thrown by Blender.

**Comment by nagadomi** [comment]
@UuuNyaa 
I guess it would be less confusing for MMD modelers if we adopt PMX Editor's vocabularies.
Also, mmd_tools is a PMX Format editing tool, and I believe that the field names related to PMX Format used by mmd_tools are English translations of the Japanese field names in PMX Editor (official software of PMX format). 
However, this is not a very strong opinion, because the meaning of the field names is understandable in the current situation.

@powroupi 
As far as I know, it affects all translatable UI parts. This may be old knowledge, but it has probably not changed.
The reason for specifying the addon's name is to unregister the dictionary set registered with that name.

File: mmd_tools/__init__.py (hunk 1 above)
Comment (UuuNyaa) [reply mmd_tools/__init__.py]: Thanks for your review. I removed the unnecessary try-except.

**Comment by UuuNyaa** [comment]
@nagadomi 

> I guess it would be less confusing for MMD modelers if we adopt PMX Editor's vocabularies.

You are also right.
At least I can say is that whichever vocabulary we adopt will be better than the current one.
If more people get confused, we can solve the problem by list or pop-up both vocabularies in the future.


@powroupi

> I'm not sure how translations affect other UI.

The following is an example of how translation affects other UIs.

image image

`Multiply` is translated into `正片叠底(相乘 A*B)` (means `Positive Overlay (Multiply A*B)`), which results in a strange display even in different contextual situations.
We cannot override this official translation without specifying the context (`text_ctxt`). 😟

**Comment by powroupi** [comment]
Okay, I see. I think we don't need to translate some common/simple vocabularies which already can be translated well by Blender such as `Apply`, `Clear`,... . Currently, It should be enough to translate only those vocabularies which would not be translated by Blender. If we want to translate other vocabularies that conflict with Blender or other addons, we should look for another way.

@UuuNyaa, let me know if your changes are done, I'll merge it then. Thank you. :smile:

**Comment by UuuNyaa** [comment]
@powroupi 
Changes are completed. Please merge it.

**Comment by powroupi** [comment]
I've merged it. Thank you.:smile:

**Pull request merged**